tests/components/thermopro/test_button.py

```diff
@@ -0,0 +1,200 @@
+    assert button.state == "unknown"
+
+
+async def test_buttons_tp358_unavailable(hass: HomeAssistant) -> None:
+    """Test tp358 set date&time button goes to unavailability."""
+    start_monotonic = time.monotonic()
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="aa:bb:cc:dd:ee:ff",
+    )
+    entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert_entity_counts(hass, Platform.BUTTON, 0, 0)
+    inject_bluetooth_service_info(hass, TP358_SERVICE_INFO)
+    await hass.async_block_till_done()
+    assert_entity_counts(hass, Platform.BUTTON, 1, 1)
+
+    button = hass.states.get("button.thermopro_tp358_4221_datetime")
+    assert button.state == "unknown"
+
+    # borrowed from bthome test_sensor.py
+    # Fastforward time without BLE advertisements
+    monotonic_now = start_monotonic + FALLBACK_MAXIMUM_STALE_ADVERTISEMENT_SECONDS + 15
+
+    with patch_bluetooth_time(monotonic_now), patch_all_discovered_devices([]):
+        async_fire_time_changed(
+            hass,
+            dt_util.utcnow()
+            + timedelta(seconds=FALLBACK_MAXIMUM_STALE_ADVERTISEMENT_SECONDS + 15),
+        )
+        await hass.async_block_till_done()
+
+    # ---
+
+    button = hass.states.get("button.thermopro_tp358_4221_datetime")
+
+    assert button.state == STATE_UNAVAILABLE
+
+
+async def test_buttons_tp358_reavailable(hass: HomeAssistant) -> None:
+    """Test TP358/TP393 set date&time button goes to unavailablity and recovers."""
+    start_monotonic = time.monotonic()
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="aa:bb:cc:dd:ee:ff",
+    )
+    entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert_entity_counts(hass, Platform.BUTTON, 0, 0)
+    inject_bluetooth_service_info(hass, TP358_SERVICE_INFO)
+    await hass.async_block_till_done()
+    assert_entity_counts(hass, Platform.BUTTON, 1, 1)
+
+    button = hass.states.get("button.thermopro_tp358_4221_datetime")
+    assert button.state == "unknown"
+
+    # borrowed from bthome test_sensor.py
+    # Fastforward time without BLE advertisements
+    monotonic_now = start_monotonic + FALLBACK_MAXIMUM_STALE_ADVERTISEMENT_SECONDS + 15
+
+    with patch_bluetooth_time(monotonic_now), patch_all_discovered_devices([]):
+        async_fire_time_changed(
+            hass,
+            dt_util.utcnow()
+            + timedelta(seconds=FALLBACK_MAXIMUM_STALE_ADVERTISEMENT_SECONDS + 15),
+        )
+        await hass.async_block_till_done()
+
+        # ---
+
+        button = hass.states.get("button.thermopro_tp358_4221_datetime")
+
+        assert button.state == STATE_UNAVAILABLE
+
+        inject_bluetooth_service_info(hass, TP358_SERVICE_INFO)
```

Comment:
As soon as I perform this action(does not matter if its inside the with block or without I get the error:
```

 File "/usr/share/hassio/homeassistant/dev/venv-core/lib/python3.13/site-packages/pytest_socket.py", line 89, in __new__
    raise SocketBlockedError()
pytest_socket.SocketBlockedError: A test tried to use socket.socket.
```
```
and
 File "/usr/share/hassio/homeassistant/dev/venv-core/lib/python3.13/site-packages/pytest_socket.py", line 252, in guarded_connect
    raise SocketConnectBlockedError(allowed_list, host)
pytest_socket.SocketConnectBlockedError: A test tried to use socket.socket.connect() with host "None" (allowed: "127.0.0.1").
```

It looks to me like I loose the bluetooth mock, but I don't understand why.

Reply:
I can't reproduce this